### PR TITLE
Fix issue #892

### PR DESCRIPTION
**Pull request by andresmmera**
See https://github.com/Qucs/qucs/issues/892

### CODE REVIEW

**Comment by guitorri** [comment]
Can you put changes in formatting a separate commit?
Right now it is hard to find and review the actual change/fix.

File: qucs/qucs-filter/filter.cpp [hunk 1]
```diff
@@ -47,195 +44,262 @@ double Filter::getE6value(double value)
...
   }
 
   QMessageBox::critical(0, "Error", "Filter type not supported.");
   return 2e30;
 }
 
 // -----------------------------------------------------------------------
-QString Filter::num2str(double Num)
-{
+QString Filter::num2str(double Num, units Unit) {
```
Comment (in3otd) [inline qucs/qucs-filter/filter.cpp]: I guess this is the function actually changed. But in the commit comment it's referenced with a different name :grin:

**Comment by andresmmera** [comment]
I added an extra argument to num2str() to handle unit. In my opinion, that seems more correct.
...It is simpler to keep num2str() untouched and simply add the meter units "m" in those lines where a TLIN is added...

I'm gonna revert these commits...

File: qucs/qucs-filter/filter.cpp (hunk 1 above)
Comment (in3otd) [reply qucs/qucs-filter/filter.cpp]: Why did you choose to fix this by passing a *Unit* to `str2num`? Some filter types already have the correct units after the value (mostly the lumped elements ones, I think) while some other don't (the ones with transmission lines mostly, I think).
The type of element used is known in advance so there's no need to pass an additional parameter, just adding the correct  unit after the prefix, like already done for the lumped elements filters, should be enough.
And while you are at it, please re-add the space between the number and the prefix (see e.g. [here](https://www.bipm.org/utils/common/pdf/si_brochure_8_en.pdf), chapter 5.3.3. And, yes, it says also that *Ohm* should actually be written *ohm* but here we will continue with the Qucs tradition for the moment...)

**Comment by in3otd** [comment]
> I'm gonna revert these commits...

just restart from the base branch and force-push, it's less work :grin:

**Comment by andresmmera** [comment]
Yes, I was doing that :-)

**Comment by in3otd** [comment]
(moving back the discussion to the PR, I commented on the issue by mistake)

>>And BTW, spaces between number and prefix/units should be used for all the units, not only ohm

> Yes, that was done by adding the space at num2str()

ah, right, I missed that and the fact that impedances do not go thru `num2str()`

**Pull request merged**